From cb66a03afc16ab38ec7d4664d3b8b77a81f7751b Mon Sep 17 00:00:00 2001
From: himta ram <hram@codeaurora.org>
Date: Wed, 29 Jan 2020 11:40:16 +0530
Subject: [PATCH 2/2] add support to check if Cs Call InProgress

Activity Manger warned intent codeaurora.telecom.action.CALL_TYPE
used for Call type in progress as a nonprotected broadcast,and this
may cause security concerns.
Remove the non-protected system broadcast with action
codeaurora.telecom.action.CALL_TYPE.
Add the isCsCallInProgress API to know the call type
in progress.

CRs-Fixed: 2676961
Change-Id: I3db21619bd8cd90b1e676fce66b1a4a6f1a5de70
---
 .../server/telecom/BluetoothPhoneServiceImpl.java   | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/com/android/server/telecom/BluetoothPhoneServiceImpl.java b/src/com/android/server/telecom/BluetoothPhoneServiceImpl.java
index 53311f93f..2292f0aa8 100644
--- a/src/com/android/server/telecom/BluetoothPhoneServiceImpl.java
+++ b/src/com/android/server/telecom/BluetoothPhoneServiceImpl.java
@@ -260,6 +260,19 @@ public class BluetoothPhoneServiceImpl {
             }
         }
 
+        @Override
+        public boolean isCsCallInProgress() {
+            boolean isCsCall = false;
+            Call activeCall = mCallsManager.getActiveCall();
+            if (mNumActiveCalls > 0) {
+                isCsCall =  ((activeCall != null) &&
+                 !(activeCall.hasProperty(Connection.PROPERTY_HIGH_DEF_AUDIO) ||
+                 activeCall.hasProperty(Connection.PROPERTY_WIFI)));
+            }
+            Log.i(TAG, "isCsCallInProgress: "+ isCsCall);
+            return isCsCall;
+        }
+
        /**
         * isHighDefCallInProgress
         * Returns true if there is any Call is in Progress with High Definition
-- 
2.28.0

